### DISCUSSION TITLE
Database connection [mariadb] not configured.

### DISCUSSION THREAD

**Discussion by blownco** [alexjustesen/speedtest-tracker#1043] [Q&A] [answered]
Hi everyone, I just installed a new install and I'm getting this error message and the container will not start. 

..+..+...............+.+.....+......+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*..+....+...+...+.....+............+...+....+...+.....+.+..+....+...+.....+......+.......+...+.....+.......+........+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*...+...+.+..+..........+..+......+......+.........+.......+..+.+..+....+.....+.........+..........+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
........+.........+...+.....+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*.............+....+...+..+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*....+....+..+....+...+........+...+......+......+...+............+....+...+........+...+.......+..+...+.......+..+...+.+.....+......+.+...............+..............+.+........+......+.+.................+.+..+.............+.....+.+...+.................+.+..+.+.....+......+.............+...+...+..+...................+...........+...+.......+........+....+...+..+.+.....+.+........+............+.+.....+.......+..+.+...+.....+.........+.......+.....+......+...+............+.......+............+.....+.+..+.+......+.....+......+.........+....+..+.............+.........+.....+.+...............+..+....+...+..+.+..+.+...........+...............+...+......+.......+..................+..+.......+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
-----
[2024-01-08 16:42:15] production.ERROR: Database connection [mariadb] not configured. {"exception":"[object] (InvalidArgumentException(code: 0): Database connection [mariadb] not configured. at /var/www/html/vendor/laravel/framework/src/Illuminate/Database/DatabaseManager.php:173)
[stacktrace]
#0 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/DatabaseManager.php(136): Illuminate\\Database\\DatabaseManager->configuration()
#1 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/DatabaseManager.php(101): Illuminate\\Database\\DatabaseManager->makeConnection()
#2 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Migrations/DatabaseMigrationRepository.php(226): Illuminate\\Database\\DatabaseManager->connection()
#3 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Migrations/DatabaseMigrationRepository.php(182): Illuminate\\Database\\Migrations\\DatabaseMigrationRepository->getConnection()
#4 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Migrations/Migrator.php(700): Illuminate\\Database\\Migrations\\DatabaseMigrationRepository->repositoryExists()
#5 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Console/Migrations/MigrateCommand.php(140): Illuminate\\Database\\Migrations\\Migrator->repositoryExists()
#6 /var/www/html/vendor/laravel/framework/src/Illuminate/Support/helpers.php(248): Illuminate\\Database\\Console\\Migrations\\MigrateCommand->Illuminate\\Database\\Console\\Migrations\\{closure}()
#7 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Console/Migrations/MigrateCommand.php(140): retry()
#8 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Console/Migrations/MigrateCommand.php(116): Illuminate\\Database\\Console\\Migrations\\MigrateCommand->repositoryExists()
#9 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Console/Migrations/MigrateCommand.php(84): Illuminate\\Database\\Console\\Migrations\\MigrateCommand->prepareDatabase()
#10 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Migrations/Migrator.php(633): Illuminate\\Database\\Console\\Migrations\\MigrateCommand->Illuminate\\Database\\Console\\Migrations\\{closure}()
#11 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Console/Migrations/MigrateCommand.php(83): Illuminate\\Database\\Migrations\\Migrator->usingConnection()
#12 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Illuminate\\Database\\Console\\Migrations\\MigrateCommand->handle()
#13 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#14 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\\Container\\Util::unwrapIfClosure()
#15 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(35): Illuminate\\Container\\BoundMethod::callBoundMethod()
#16 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(662): Illuminate\\Container\\BoundMethod::call()
#17 /var/www/html/vendor/laravel/framework/src/Illuminate/Console/Command.php(211): Illuminate\\Container\\Container->call()
#18 /var/www/html/vendor/symfony/console/Command/Command.php(326): Illuminate\\Console\\Command->execute()
#19 /var/www/html/vendor/laravel/framework/src/Illuminate/Console/Command.php(180): Symfony\\Component\\Console\\Command\\Command->run()
#20 /var/www/html/vendor/symfony/console/Application.php(1096): Illuminate\\Console\\Command->run()
#21 /var/www/html/vendor/symfony/console/Application.php(324): Symfony\\Component\\Console\\Application->doRunCommand()
#22 /var/www/html/vendor/symfony/console/Application.php(175): Symfony\\Component\\Console\\Application->doRun()
#23 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(201): Symfony\\Component\\Console\\Application->run()
#24 /var/www/html/artisan(35): Illuminate\\Foundation\\Console\\Kernel->handle()
#25 {main}
"} 
s6-rc: warning: unable to start service laravel-automations: command exited 1
/run/s6/basedir/scripts/rc.init: warning: s6-rc failed to properly bring all the services up! Check your logs (in /run/uncaught-logs/current if you have in-container logging) for more information.
prog: fatal: stopping the container.
--------------------------------------------------------------------

 _______                       __ __                __
|     __|.-----.-----.-----.--|  |  |_.-----.-----.|  |_
|__     ||  _  |  -__|  -__|  _  |   _|  -__|__ --||   _|
|_______||   __|_____|_____|_____|____|_____|_____||____|
         |__|
 _______                   __
|_     _|.----.---.-.----.|  |--.-----.----.
  |   |  |   _|  _  |  __||    <|  -__|   _|
  |___|  |__| |___._|____||__|__|_____|__|

--------------------------------------------------------------------

To support Speedtest Tracker visit:
https://github.com/alexjustesen/speedtest-tracker
-------------------------------------
GID/UID
-------------------------------------

User uid:    1000
User gid:    1000
-------------------------------------

🔒 SSL_MODE has been set to MIXED, setting the web server to work in HTTP + HTTPS...
🏃‍♂️ An SSL key was not detected, so I'll generate a self-signed SSL certificate pair for you...

🐇  Configuring Speedtest Tracker...

🔒  Fixing app path file permissions...
✅  Permissions fixed.

🧹  Clearing any previous caches...
✅  Cache cleared.

✅  Environment file exists.

🔗  Creating symlinks for config and log files...
✅  Symlinks created.

⏳  Generating an application key...
⚠️  An application key was generated at start up, no environment variable was set.
👀  To set an application key that persists, read the docs: https://docs.speedtest-tracker.dev/

💰  Building the cache...
✅  Cache set.

🚛  Migrating the database...

In DatabaseManager.php line 173:
                                                 
  Database connection [mariadb] not configured.  
                                                 

** Press ANY KEY to close this window **

**Comment by alexjustesen** [comment]
Can you share your `docker-compose.yml` file?

**Reply by blownco** [reply]
this was installed from unraid community apps. I can  share those settings?

**Reply by blownco** [reply]
https://photos.google.com/lr/photo/ACEPBf3bqdcIIAaTtVROiMJgXCWbeCK86DXTEW-3Q_Jv6_GsZV9fSolh_g6DnIRqX7RMKd4agzg428kBEvLPV4MVGBDhhqf2Xg

**Reply by alexjustesen** [reply, marked as answer]
The Unraid app assumes you have an external DB already, like a MySQL or MariaDB container running or inside a VM. You'll need to set that up first and then you can add the `DB_` [environment variables](https://docs.speedtest-tracker.dev/getting-started/installation/installation) to the "Database" section.

![image](https://github.com/alexjustesen/speedtest-tracker/assets/1144087/efe486ad-6540-4010-8b77-f79025c9f36a)

**Reply by blownco** [reply]
I got it working thanks for the help!
